tensorhive/core/utils/reservation_allowed.py

```diff
@@ -0,0 +1,78 @@
+from datetime import time, timedelta
+from sqlalchemy.orm.exc import NoResultFound
+from tensorhive.models.Resource import Resource
+
+
+def check_schedules(start_date, end_date, schedules):
+    reservation_allowed = False
+    while True:
+        start_date_changed = False
+        for schedule in schedules:
+            day = start_date.weekday() + 1
+            if str(day) in schedule.schedule_days and schedule.hour_start <= start_date.time():
+                if schedule.hour_end == time(hour=23, minute=59):
+                    start_date = start_date.replace(hour=0, minute=0) + timedelta(days=1)
+                elif start_date.time() < schedule.hour_end:
+                    start_date = start_date.replace(hour=schedule.hour_end.hour, minute=schedule.hour_end.minute)
+                else:
+                    continue
+                start_date_changed = True
+                if start_date >= end_date:
+                    reservation_allowed = True
+                    break
+        if reservation_allowed or not start_date_changed:
+            break
+    return start_date
+
+
+def is_reservation_allowed(user, reservation):
+    try:
+        resource = Resource.get(reservation.protected_resource_id)
+    except NoResultFound:
+        return False
+
+    user_restrictions = user.get_restrictions(include_group=True)
+    # get global restrictions or applied to selected resource
+    restrictions = []
+    for restriction in user_restrictions:
+        if restriction.is_global or resource in restriction.resources:
+            restrictions.append(restriction)
```

Comment:
Optional really, but it can be simplified:
```suggestion
    restrictions = [r for r in user_restrictions if r.is_global or resource in r.resources]
```